core/androidNative/src/internal/TzdbBionic.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019-2024 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+/*
+ * Based on the bionic project.
+ * Copyright (C) 2017 The Android Open Source Project
+ */
+
+package kotlinx.datetime.internal
+
+private class TzdbBionic(private val rules: Map<String, Entry>) : TimeZoneDatabase {
+    override fun rulesForId(id: String): TimeZoneRules =
+        rules[id]?.readRules() ?: throw IllegalStateException("Unknown time zone $id")
+
+    override fun availableTimeZoneIds(): Set<String> = rules.keys
+
+    class Entry(val file: ByteArray, val offset: Int, val length: Int) {
+        fun readRules(): TimeZoneRules = readTzFile(file.copyOfRange(offset, offset + length)).toTimeZoneRules()
+    }
+}
+
+// see https://android.googlesource.com/platform/bionic/+/master/libc/tzcode/bionic.cpp for the format
+internal fun TzdbBionic(): TimeZoneDatabase = TzdbBionic(buildMap<String, TzdbBionic.Entry> {
+    for (path in listOf(
+        Path.fromString("/system/usr/share/zoneinfo/tzdata"), // immutable fallback tzdb
+        Path.fromString("/apex/com.android.tzdata/etc/tz/tzdata"), // an up-to-date tzdb, may not exist
+    )) {
```

Comment:
Shouldn't the immutable fallback be last in list? @dkhalanskyjb

Reply:
I see, sorry for bothering then! It just felt wrong, I'm used to see 'fallbacks' at the end of lists. Thanks for working on this 🙏